### ISSUE TITLE
"paper.md" found twice, even though there is only one

### ISSUE THREAD

**Issue by PABourdin** [openjournals/whedon#80]
Using the webpage for looking at a paper preview:
https://whedon.theoj.org/
I can give the correct URL to our repository that contains *exactly* one "paper.md":
https://github.com/pencil-code/pencil-code
But the compilation aborts after some time with a wrong error message that there would be multiple "paper.md" files.

Just we have our "paper" directory as a subdirectory of "doc", which we require to have there because of our code structure:
pencil-code/doc/paper/paper.md

If I give instead this URL to the preview tool:
https://github.com/pencil-code/pencil-code/doc
I have the same result as if I would give a non-existent URL, see issue #79

**Comment by arfon** [comment]
This is failing because Whedon looks for files named `paper.md` and `paper.tex` and you have one of each (https://github.com/pencil-code/pencil-code/blob/master/misc/f90-timings/doc/paper.tex, https://github.com/pencil-code/pencil-code/blob/master/doc/paper/paper.md).

Ultimately we will need to find a way to allow people to specify which paper they want to compile. For now though this isn't possible sorry.

One potential short term fix is for you to rename https://github.com/pencil-code/pencil-code/blob/master/misc/f90-timings/doc/paper.tex to something other than `paper.tex`

**Comment by PABourdin** [comment]
@arfon 
Yes, being able to choose the file name, or better the full path to the *.md file, is a "must have" if you want your application to be usable for established software project that will probably already have several "paper.tex" or "paper.md" in place, somewhere.

So yes, please consider this as a strong support to your comment.

**Comment by arfon** [comment]
We also support alternative branches so you can always create a `joss-paper` branch for your project and have the JOSS paper be the only `paper.md` in the source code on that branch.

**Comment by PABourdin** [comment]
@arfon 
Thanks for the hint, but we do not use branches that would become "stale" in our case very quickly...

**Comment by arfon** [comment]
> Thanks for the hint, but we do not use branches that would become "stale" in our case very quickly...

OK, well I've given you a few suggested solutions here. We don't have a lot of capacity to work on improving the preview service right now sorry so if this is a blocker for you to submit, you'll need to use one one of the suggestions I've offered here.

**Comment by arfon** [comment]
/ cc @tarleb - this is an example of where the Docker-powered author preview tool could help :-)

**Comment by PABourdin** [comment]
@arfon 
Sorry, please do not get me wrong, here. I have already fixed the issue on our side as you suggested. I'm just telling that having a proper way to define the filename, or alternatively the subdirectory to search for "paper.md", as you suggested, is what I would also strongly recommend to do. - From the perspective of the owners of large scientific codes with history.
